docs/build/guides/conventions/soroban-contract-init-template.mdx

```diff
@@ -0,0 +1,432 @@
+
+```bash
+stellar contract init `hello_contract` -w hello -f https://github.com/CaptainPrinz/soroban-react-template.git
+```
+
+The snippet above initializes a new Stellar smart contracts project, `hello_contract`, that uses a prebuilt `hello_world` contract and adds the frontend template we deployed to a remote repository.
+
+### Setting the environment variables
+
+Our templates offer a great deal of customization via environment variables. These variables need to be set by the user by creating a `.env` file otherwise the process will fail as these variables will be undefined.
+
+### Starting the script
+
+On your terminal run the following code:
+
+```bash
+npm run setup
+```
+
+This runs the `initialize.js` script and first sets the network profile, account identities, builds the contract `.wasm` file, deploys the contracts, and generates the bindings.
+
+If successful, there should be the following changes in the directory:
+
+1. A new `.soroban` directory, which contains the network profile files, account identity files, and hosted smart contract files in json format
+2. A `packages` directory containing the generated TypeScript bindings
+3. A `contracts.json` file
+
+You are now finally able to import these generated bindings in your frontend application, create their clients, and use them to interact with the contracts deployed on the network.
+
+##Interacting with the contract bindings
+After running the initialization script, there should be new `packages` directory which confirms successful bindings generation. Navigate to the `packages` directory and there should be a package for the deployed contract(s).
+This package will form the last piece in solving the puzzle of interacting with soroban contracts from frontend appications which will be discussed briefly.
+
+To work with the bindings generated, it is recommended the wrap the bindings by creating a new module, instantiating a new client, then finally importing the client and invoking the contract functions.
+
+While it is not necessary to follow the procedure above, you stand to have a better code structure and a much easier way of fixing the code should anything go wrong.
+
+However given the simple structure of our contract, we will proceed to integrate it directly in the frontend without creating a wrapper module.
+
+##Setting up the hello world package
+Navigate to the packages folder to find the hello wold package and observe the directory contents. If the directory does not contain a `dist` folder, you must build the hello world typescript bindings for the package to work in our react app.
+
+```bash
+npm run build
+```
+
+After running the `build` command, there should now be a `dist` directory containing `index.js` and `index.ts` files. These will serve as an entry point to the package when it is eventually imported.
```

Comment:
I'm not sure the `build` script actually does what you want it to here. In your template repository, `build` is just calling `vite build`, which only concerns itself with the React app, not the contracts themselves.

Reply:
No In this case I'm referring to the 'build' command in the typescript binding package for the contract. I f you were to observe it's package.json file you'll find the build command

Reply:
Ah! I see. I must have missed the bit about going to the `packages/hello_world` before this command. Thanks!